src/Compilers/CSharp/Portable/Emitter/EditAndContinue/CSharpSymbolMatcher.cs

```diff
@@ -718,6 +728,12 @@ private bool AreNamedTypesEqual(NamedTypeSymbol type, NamedTypeSymbol other)
                 // TODO: Test with overloads (from PE base class?) that have modifiers.
                 Debug.Assert(!type.HasTypeArgumentsCustomModifiers);
                 Debug.Assert(!other.HasTypeArgumentsCustomModifiers);
+
+                if (type.IsTupleType)
+                {
+                    return other.IsTupleType && type.TupleElementTypes.SequenceEqual(other.TupleElementTypes, AreTypesEqual);
+                }
+
                 return type.TypeArgumentsNoUseSiteDiagnostics.SequenceEqual(other.TypeArgumentsNoUseSiteDiagnostics, AreTypesEqual);
```

Comment:
Is it possible `AreNamedTypesEqual` will be called with two types where `type.Name == other.Name && !type.IsTupleType && other.IsTupleType`? If so, we'll treat the types as equal if the two have the same type arguments.